### PR TITLE
Adhesive mascara: refactor team users + popovers

### PR DESCRIPTION
**Pull request by modernserf**
## Links
* Remix link: https://adhesive-mascara.glitch.me

## GIF/Screenshots
removing other user as admin
Screen Shot 2019-05-13 at 4 27 36 PM
removing self
Screen Shot 2019-05-13 at 4 29 10 PM
removing other user from projects
Screen Shot 2019-05-13 at 4 30 28 PM
deleting the team
Screen Shot 2019-05-13 at 4 33 50 PM
janky css that will be fixed later
Screen Shot 2019-05-13 at 4 38 20 PM

## Changes
* replace `{...props}` spreads with explicit props
* combine team users presenters into single presenter
* move invitees state into team users
* hookify DeleteTeamPop
* hookify TeamUserRemovePop

## How To Test
* log into https://adhesive-mascara.glitch.me
* create a new team
* add a user to the team (the CSS for pending users is janky, but that's also the case on prod; that's gonna be fixed in the next PR)
* add a project to the team
* have the other user join that project
* remove the user from the team
* delete the team

## Future PRs
- [ ] componentize team users
- [ ] fix janky CSS

### CODE REVIEW

File: src/presenters/includes/team-users.js [hunk 1]
```diff
@@ -199,8 +187,54 @@ AddTeamUser.defaultProps = {
...
+  }
+});
+
+const TeamUserContainer = ({ team, removeUserFromTeam, updateUserPermissions, updateWhitelistedDomain, inviteEmail, inviteUser, joinTeam }) => {
+  const { currentUser } = useCurrentUser();
+  const currentUserIsOnTeam = userIsOnTeam({ team, user: currentUser });
+  const currentUserIsTeamAdmin = userIsTeamAdmin({ team, user: currentUser });
+  const currentUserCanJoinTeam = userCanJoinTeam({ team, user: currentUser });
+  const { value } = useInvitees(team, currentUserIsOnTeam);
+  const invitees = value || [];
```
Comment (sarahzinger) [inline src/presenters/includes/team-users.js]: are there circumstances where we would hit this ` || []`? looks like it's already captured in useInvitees right?

File: src/presenters/includes/team-users.js [hunk 2]
```diff
@@ -199,8 +187,54 @@ AddTeamUser.defaultProps = {
...
   <button className="button button-small button-cta join-team-button" onClick={onClick}>
     Join Team
   </button>
 );
+
+const useInvitees = createAPIHook(async (api, team, currentUserIsOnTeam) => {
+  if (!currentUserIsOnTeam) return [];
+
+  try {
+    const data = await Promise.all(team.tokens.map(({ userId }) => api.get(`users/${userId}`)));
```
Comment (sarahzinger) [inline src/presenters/includes/team-users.js]: is this using the old api? should we switch?

File: src/presenters/pop-overs/delete-team-pop.js [hunk 3]
```diff
@@ -1,88 +1,62 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import { withRouter } from 'react-router-dom';
 import Loader from 'Components/loader';
+// import { teamAdmins } from 'Models/team';
```
Comment (sarahzinger) [inline src/presenters/pop-overs/delete-team-pop.js]: is this comment intentional?

Comment (sarahzinger) [reply src/presenters/pop-overs/delete-team-pop.js]: oh I see the comment below my bad

File: src/presenters/includes/team-users.js (hunk 1 above)
Comment (modernserf) [reply src/presenters/includes/team-users.js]: when it hasn't loaded yet

File: src/presenters/includes/team-users.js (hunk 2 above)
Comment (modernserf) [reply src/presenters/includes/team-users.js]: probably -- i'll test to see if all the data we need is in the new API

File: src/presenters/includes/team-users.js (hunk 1 above)
Comment (sarahzinger) [reply src/presenters/includes/team-users.js]: ahhh gotcha thanks!

File: src/presenters/includes/team-users.js (hunk 2 above)
Comment (modernserf) [reply src/presenters/includes/team-users.js]: yup, new API seems to work

**Pull request merged**